src/test/java/com/pgssoft/httpclient/HttpClientMockExceptionFallthroughTest.java

```diff
@@ -0,0 +1,71 @@
+package com.pgssoft.httpclient;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class HttpClientMockExceptionFallthroughTest {
+
+    @Test
+    void runtimeExceptionShouldFallThroughUnchanged() {
```

Comment:
Please make this test more concise. Please look at the example test from my first comment.